synapse/handlers/_base.py

```diff
@@ -133,10 +133,9 @@ def handle_new_client_event(self, event, context, extra_destinations=[],
         for k, s in context.current_state.items():
             try:
                 if k[0] == EventTypes.Member:
-                    if s.content["membership"] == Membership.JOIN:
```

Comment:
Right. If the event about to be sent (``event``) is an ``m.room.member`` event then you want to add the domain of the affected member (``state_key``) to the ``destinations``. 

Unsure what ``k, s`` are in this; I would've assumed ``context.current_state.items()`` == A list of the current state events for the room, but there are 2 values from this which are being used interchangeably it seems.